src/azul/openapi/validation.py

```diff
@@ -0,0 +1,198 @@
+            spec_schema = self.content['application/json']['schema']
+            try:
+                filter_json = json.loads(request_parameter['value'])
+            except Exception:
+                raise BadRequestError(f"The '{request_parameter['name']}' parameter is not valid JSON")
+            if type(filter_json) is not dict:
+                raise BadRequestError(f"The '{request_parameter['name']}' parameter must be a dictionary.")
+            for facet, filter_ in filter_json.items():
+                try:
+                    spec_property = spec_schema['properties'][facet]
+                    try:
+                        relation, value = one(filter_.items())
+                    except Exception:
+                        raise BadRequestError(
+                            f"The 'filters' parameter entry for '{filter_}' must be a single-item dictionary.")
+                    else:
+                        try:
+                            relation_spec = one(x for x in spec_property['oneOf'] if relation in x['properties'])
+                            relation_type = relation_spec['properties'][relation]['type']
+                        except ValueError:
+                            valid_relations = [list(x['properties'].keys())[0] for x in spec_property['oneOf']]
+                            raise BadRequestError(f"The relation in the '{request_parameter['name']}' parameter entry"
+                                                  f" for '{facet}' must be one of '{valid_relations}'.")
+                        if not isinstance(value, schema.type_lookup(relation_type)):
+                            raise BadRequestError(
+                                msg=f"The value of the '{relation}' relation in the '{request_parameter['name']}'"
+                                    f" parameter entry for '{facet}' is not a {relation_type}.")
+                        if relation in ("contains", "within", "intersects"):
+                            min_items = relation_spec['properties'][relation]['minItems']
+                            max_items = relation_spec['properties'][relation]['maxItems']
+                            try:
+                                for nested_value in value:
+                                    assert min_items <= len(nested_value) <= max_items
+                                    assert isinstance(nested_value, schema.type_lookup(relation_type))
+                            except (AssertionError, TypeError):
+                                raise BadRequestError(
+                                    msg=f"The value of the '{relation}' relation in the '{request_parameter['name']}'"
+                                        f" parameter entry for '{facet}' does not match the spec '{relation_spec}'.")
+
+                except KeyError:
+                    raise BadRequestError(f"Unknown facet '{facet}'")
+
+        if getattr(self, "content") is not None:
```

Comment:
quotes
inner function just for if is bad
what does 'content' have to do with filters?